File: exercices/solutions/smileys.f90
PROGRAM smileys
use mpi
IMPLICIT NONE

INTEGER ierr, rank, size

CALL MPI_Init(ierr)

CALL MPI_Comm_rank (MPI_COMM_WORLD, &
                    rank, ierr)

CALL MPI_Comm_size (MPI_COMM_WORLD, &
                    size, ierr)
IF (rank == 0) THEN
    WRITE(*,*) rank, '/', size, '  :-|'
ENDIF
IF (rank == 1) THEN
    WRITE(*,*) rank, '/', size, '  :-)'
ENDIF
IF (rank == 2) THEN
    WRITE(*,*) rank, '/', size, '  :-D'
ENDIF
IF (rank == 3) THEN
    WRITE(*,*) rank, '/', size, '  :-P'
ENDIF

CALL  MPI_Finalize(ierr)
END PROGRAM smileys

File: exercices/solutions/smileys.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[])
{
  int rank, size;

  MPI_Init( &argc, &argv );

  MPI_Comm_rank( MPI_COMM_WORLD, &rank );
  MPI_Comm_size( MPI_COMM_WORLD, &size );

  if (rank == 0) printf( "%3d/%-3d   :-|\n", rank, size );
  if (rank == 1) printf( "%3d/%-3d   :-)\n", rank, size );
  if (rank == 2) printf( "%3d/%-3d   :-D\n", rank, size );
  if (rank == 3) printf( "%3d/%-3d   :-P\n", rank, size );

  MPI_Finalize();

  return 0;
}
